### PR TITLE
Check gnome2/3 proxy mode before returning config values.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: config_gnome2.c [hunk 1]
```diff
@@ -34,23 +34,29 @@ typedef struct g_proxy_config_gnome2_s {
...
+    bool equal = false;
 
-    mode =
+    char *compare_mode =
         g_proxy_config_gnome2.gconf_engine_get_string(g_proxy_config_gnome2.gconf_default, "/system/proxy/mode", NULL);
-    if (mode) {
-        auto_discover = strcmp(mode, "auto");
-        g_proxy_config_gnome2.g_free(mode);
+    if (compare_mode) {
+        equal = strcmp(compare_mode, mode);
```
Comment (sergio-nsk) [inline config_gnome2.c:43]: Isn't an error here? Shouldn't be `equal = strcmp(compare_mode, mode) == 0;`?

File: config_gnome2.c [hunk 2]
```diff
@@ -34,23 +34,29 @@ typedef struct g_proxy_config_gnome2_s {
...
 g_proxy_config_gnome2_s g_proxy_config_gnome2;
 
-bool proxy_config_gnome2_get_auto_discover(void) {
-    char *mode = NULL;
-    bool auto_discover = false;
+static bool proxy_config_gnome2_is_mode(const char *mode) {
+    bool equal = false;
 
-    mode =
+    char *compare_mode =
```
Comment (sergio-nsk) [inline config_gnome2.c:40]: IMHO `compare_mode` is not a convenient. I would use `set_mode`. Or woild swap names at lease, `compare_mode` to the parameter name.

File: config_gnome2.c (hunk 1 above)
Comment (nmoinvaz) [reply config_gnome2.c]: Fixed. Nice catch.

File: config_gnome2.c (hunk 2 above)
Comment (nmoinvaz) [reply config_gnome2.c]: Fixed.

**Comment by nmoinvaz** [comment]
CI tests fixed.

**Pull request merged**